src/PAModel/CanvasDocument.cs

```diff
@@ -351,17 +351,23 @@ internal void ApplyAfterMsAppLoadTransforms(ErrorContainer errors)
 
             // Shard templates, parse for default values
             var templateDefaults = new Dictionary<string, ControlTemplate>();
-            foreach (var template in _templates.UsedTemplates)
```

Comment:
You could make this simpler if nullable by using the question mark to check for null
 `foreach (var template in _templates?.UsedTemplates)`

Reply:
Same suggestion in almost all checks below